gnat2goto/driver/tree_walk.adb

```diff
@@ -1419,6 +1423,80 @@ package body Tree_Walk is
 
    end Do_Loop_Statement;
 
+   ---------------
+   -- Do_Pragma --
+   ---------------
+
+   procedure Do_Pragma (N : Node_Id; Block : Irep) is
+
+      --------------------------------
+      -- Do_Pragma_Assert_or_Assume --
+      --------------------------------
+
+      procedure Do_Pragma_Assert_or_Assume
+        (N : Node_Id; Block : Irep; Which : String);
+
+      procedure Do_Pragma_Assert_or_Assume
+        (N : Node_Id; Block : Irep; Which : String)
+      is
+         A_Irep : constant Irep := (if Which = "assert"
+                                    then New_Irep (I_Code_Assert)
+                                    else New_Irep (I_Code_Assume));
+
+         --  To be set by iterator:
+         Check : Irep := Ireps.Empty;
+
+         ----------------
+         -- Handle_Arg --
+         ----------------
+
+         procedure Handle_Arg
+           (Arg_Pos : Positive; Arg_Name : String; Expr : Node_Id);
+
+         procedure Handle_Arg
+           (Arg_Pos : Positive; Arg_Name : String; Expr : Node_Id) is
+         begin
+            if Arg_Name = "check" or (Arg_Name = "" and Arg_Pos = 1) then
+               Check := Do_Expression (Expr);
+            elsif Arg_Name = "message" or (Arg_Name = "" and Arg_Pos = 2) then
+               null; -- ignore, since assert irep has no msg
```

Comment:
You're right, they don't at the moment. I'm saying long-term we should improve our source location representation so that we can use https://github.com/diffblue/cbmc/blob/master/src/util/source_location.h#L68 rather than simply proxying GNAT's source locations. No need to change anything in this PR.